internal/cmd/argus/grafana/single-sign-on/disable/disable_test.go

```diff
@@ -0,0 +1,265 @@
+	request := testClient.UpdateGrafanaConfigs(testCtx, testInstanceId, testProjectId)
+	for _, mod := range mods {
+		mod(&request)
+	}
+	return request
+}
+
+func TestParseInput(t *testing.T) {
+	tests := []struct {
+		description   string
+		flagValues    map[string]string
+		isValid       bool
+		expectedModel *inputModel
+	}{
+		{
+			description:   "base",
+			flagValues:    fixtureFlagValues(),
+			isValid:       true,
+			expectedModel: fixtureInputModel(),
+		},
+		{
+			description: "no values",
+			flagValues:  map[string]string{},
+			isValid:     false,
+		},
+		{
+			description: "project id missing",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				delete(flagValues, projectIdFlag)
+			}),
+			isValid: false,
+		},
+		{
+			description: "project id invalid 1",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				flagValues[projectIdFlag] = ""
+			}),
+			isValid: false,
+		},
+		{
+			description: "project id invalid 2",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				flagValues[projectIdFlag] = "invalid-uuid"
+			}),
+			isValid: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.description, func(t *testing.T) {
+			cmd := NewCmd(nil)
+			err := globalflags.Configure(cmd.Flags())
+			if err != nil {
+				t.Fatalf("configure global flags: %v", err)
+			}
+
+			for flag, value := range tt.flagValues {
+				err := cmd.Flags().Set(flag, value)
+				if err != nil {
+					if !tt.isValid {
+						return
+					}
+					t.Fatalf("setting flag --%s=%s: %v", flag, value, err)
+				}
+			}
+
+			err = cmd.ValidateRequiredFlags()
+			if err != nil {
+				if !tt.isValid {
+					return
+				}
+				t.Fatalf("error validating flags: %v", err)
+			}
+
+			model, err := parseInput(cmd)
+			if err != nil {
+				if !tt.isValid {
+					return
+				}
+				t.Fatalf("error parsing flags: %v", err)
+			}
+
+			if !tt.isValid {
+				t.Fatalf("did not fail on invalid input")
+			}
+			diff := cmp.Diff(model, tt.expectedModel)
+			if diff != "" {
+				t.Fatalf("Data does not match: %s", diff)
+			}
+		})
+	}
+}
+
+func TestBuildRequest(t *testing.T) {
+	tests := []struct {
+		description            string
+		model                  *inputModel
+		getGrafanaConfigsFails bool
+		getGrafanaConfigsResp  *argus.GrafanaConfigs
+		isValid                bool
+		expectedRequest        argus.ApiUpdateGrafanaConfigsRequest
+	}{
+		{
+			description:            "base",
+			model:                  fixtureInputModel(),
+			getGrafanaConfigsFails: false,
+			getGrafanaConfigsResp:  fixtureGrafanaConfigs(),
+			isValid:                true,
+			expectedRequest: fixtureRequest().UpdateGrafanaConfigsPayload(
+				argus.UpdateGrafanaConfigsPayload{
+					GenericOauth:     argusUtils.ToPayloadGenericOAuth(fixtureGrafanaConfigs().GenericOauth),
+					PublicReadAccess: fixtureGrafanaConfigs().PublicReadAccess,
+					UseStackitSso:    utils.Ptr(false),
+				}),
```

Comment:
Shouldn't this be the fixture request? That way the base case is simpler

Reply:
I'll leave the comment to get your thoughts on it, but thinking about it maybe this is better here because of the calls to argus utils.

Reply:
Totally agree, good suggestion